components/ai-services-mgt/org.wso2.carbon.ai.service.mgt/src/main/java/org/wso2/carbon/ai/service/mgt/token/AIAccessTokenManager.java

```diff
@@ -0,0 +1,265 @@
+        return this.accessToken;
+    }
+
+    private AccessTokenRequestHelper createDefaultHelper() {
+
+        return new AccessTokenRequestHelper(LOGIN_FLOW_AI_KEY, LOGIN_FLOW_AI_TOKEN_ENDPOINT,
+                HttpAsyncClients.createDefault());
+    }
+
+    /**
+     * Set the client ID.
+     *
+     * @param clientId The client ID.
+     */
+    public void setClientId(String clientId) {
+
+        this.clientId = clientId;
+    }
+
+    /**
+     * Get the client ID.
+     *
+     * @return The client ID.
+     */
+    public String getClientId() {
+
+        return this.clientId;
+    }
+
+    /**
+     * Helper class to request access token from the Login Flow AI service.
+     */
+    protected static class AccessTokenRequestHelper {
+
+        private final CloseableHttpAsyncClient client;
+        private final Gson gson;
+        private final String key;
+        private final String aiServiceTokenEndpoint;
+        private static final int MAX_RETRIES = IdentityUtil.getProperty(
+                "AIServices.LoginFlow.TokenRequestMaxRetries") != null ?
+                Integer.parseInt(IdentityUtil.getProperty("AIServices.LoginFlow.TokenRequestMaxRetries")) : 3;
+        private static final long TIMEOUT = IdentityUtil.getProperty(
+                "AIServices.LoginFlow.TokenRequestTimeout") != null ?
+                Long.parseLong(IdentityUtil.getProperty("AIServices.LoginFlow.TokenRequestTimeout")) : 3000;
+
+        AccessTokenRequestHelper(String key, String tokenEndpoint, CloseableHttpAsyncClient client) {
+
+            this.client = client;
+            this.gson = new GsonBuilder().create();
+            this.key = key;
+            this.aiServiceTokenEndpoint = tokenEndpoint;
+        }
+
+        /**
+         * Request access token to access the Login Flow AI service.
+         *
+         * @return the JWT access token.
+         * @throws AIServerException If an error occurs while requesting the access token.
+         */
+        public String requestAccessToken() throws AIServerException {
+
+            String tenantDomain = getTenantDomainFromContext();
+            LOG.info("Initiating access token request for Login Flow AI service from tenant: " + tenantDomain);
+            try {
+                client.start();
+                for (int attempt = 0; attempt < MAX_RETRIES; attempt++) {
+                    HttpPost post = new HttpPost(aiServiceTokenEndpoint);
+                    post.setHeader("Authorization", "Basic " + key);
+                    post.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+                    StringEntity entity = new StringEntity("grant_type=client_credentials");
+                    entity.setContentType(new BasicHeader(HTTP.CONTENT_TYPE, "application/x-www-form-urlencoded"));
+                    post.setEntity(entity);
+
+                    CountDownLatch latch = new CountDownLatch(1);
+                    final String[] accessToken = new String[1];
```

Comment:
***NOTE: What should we do in multi node setup, should we use multiple JWT tokens or persist the access token in DB?